frontend/src/utills/index.ts

```diff
@@ -0,0 +1,58 @@
+export * from "./graph";
+
+/**
+ * Formats a given percentage value.
+ * @param {number} numerator - The numerator of the percentage.
+ * @param {number} denominator - The denominator of the percentage.
+ * @returns {string} - The formatted percentage string.
+ */
+export function getPercent(numerator: number, denominator: number): string {
+  if (denominator === 0) {
+    return "0%";
+  }
+  const percent = (numerator / denominator) * 100;
+  return `${percent.toFixed(1)}%`;
+}
+
+/**
+ * Formats a given millisecond value into a human-readable time format.
+ * @param {number} milliseconds - The value in milliseconds to be formatted.
+ * @returns {string} - The formatted time string.
+ */
+export function filterMillisecond(milliseconds: number): string {
```

Comment:
This function is too complicated to write， Please use the package: pretty-ms

https://www.npmjs.com/package/pretty-ms